src/recipe/mod.rs

```diff
@@ -23,6 +23,12 @@ pub trait Recipe {
         signal: &Signal,
     ) -> Result<()>;
     fn run(&self, contract: &Contract, build_cmd: String, signal: &Signal) -> Result<()>;
-    fn run_build(&self, contract: &Contract, config: BuildConfig, signal: &Signal) -> Result<()>;
+    fn run_build(
+        &self,
+        contract: &Contract,
+        config: BuildConfig,
+        signal: &Signal,
+        build_args_opt: Option<Vec<String>>,
```

Comment:
Why not just accept `build_args: Vec<String>`?

Reply:
When calling, run_build is not a parameter that must be passed in, so here I use Option. From an engineering point of view, it is more reasonable to use ```Vec<string>``` directly.
Do you think it needs to be change?

Reply:
OK, just fine here.